libs/klv/src/main/java/org/codice/alliance/libs/klv/ObjectCountryCodesHandler.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.libs.klv;
+
+import ddf.catalog.data.Attribute;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+import org.apache.tika.parser.txt.CharsetDetector;
+import org.apache.tika.parser.txt.CharsetMatch;
+import org.codice.ddf.libs.klv.KlvDataElement;
+import org.codice.ddf.libs.klv.data.raw.KlvBytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This handler expects a KlvBytes object which should hold the country code value. It detects the
```

Comment:
✏️ If anyone else is curious, this works because Java Strings are encoding-agnostic. As a result, when specifying the encoding in the constructor, the String is initialized correctly, and then it follows the behavior of our platform-default, UTF-8.

Reply:
I didn't know this until working on this issue. Interesting stuff!

Reply:
What does encoding-agnostic mean?